### ISSUE TITLE
Incorrect number of statements in python 3.8.0

### ISSUE THREAD

**Issue by ElPincheTopo** [coveragepy/coveragepy#861]
**Describe the bug**
Coverage gives different results in python 3.8.0 than what it did in 3.7.4
Output from Python 3.7.4:

 Name | Stmts | Miss | Branch | BrPart | Cover
------|-------|------|--------|--------|-------
 TOTAL | 5078 | 511 | 1251 | 132 | 87%

 245 passed, 36 warnings

Output from Python 3.8.0:

 Name | Stmts | Miss | Branch | BrPart | Cover
------|-------|------|--------|--------|-------
 TOTAL | 7981 | 3737 | 1482 | 588 | 53%

 245 passed, 36 warnings

It seems like in 3.8 statements are being counted more than once.

**To Reproduce**
I'm having this issue in a private repo that I'm not allowed to share and I haven't been able to isolate the problem to just one file to be able to reproduce it in a different code base.

I'm doing this in Python 3.7.4 and 3.8.0
Using coverage-4.5.1, pytest-5.2.1 pytest-cov-2.7.1
I'm also using hypothesis-4.41.3 if that helps.

I get the same number of statements with out without branch = True and show_missing = True.

I get the same result if pytest prints the coverage report to the terminal than using coverage run and then coverage report.

**Comment by nedbat** [comment]
I'm not surprised that the numbers might change a little bit from version to version, but these numbers are changing a lot.

The first thing to do is try using the latest coverage (4.5.4) or even the latest alpha of 5.0 (5.0a8).  There are fixes for 3.8 in each of them that are not in 4.5.1

If that doesn't clear it up, can you compare the specific results for the same file in the two versions to see how the lines are categorized differently? In the HTML report, you can turn on the green line markers to get a better idea.

**Comment by ElPincheTopo** [comment]
Thanks for the comment @nedbat I'll try the latest version of the library and se if that fixes it.
I also noticed that the HTML report is not displaying the missed things. I can see the green and yellow lines, but it's not displaying the red lines.

**Comment by nedbat** [comment]
I don't understand why the HTML report would be different than the text report.  Did you try the buttons at the top of the page, like these:
<img width="366" alt="image" src="https://user-images.githubusercontent.com/23789/67484345-ea882000-f635-11e9-983e-202237a06d76.png">
"8 missing" is clickable, and toggles the red lines on and off.

**Comment by ElPincheTopo** [comment]
I tested again with python 3.8 but this time using the latest version of coverage and everything is working fine again. Sorry for not checking this in advance.

When I created this issue I thought I was using the latest version of coverage but it was just the latest one I had access to on an offline system.

Just to clarify the issue with the html report in case is relevant, the square saying 8 missing is showing the same number as the text report. But no lines are actually marked red. I did try the buttons at the top. Even without clicking the button I should be able to see the red color on the left margin but it's not showing.

I'm closing this now as it is not an issue on the latest version.
